.github/workflows/test.yaml

```diff
@@ -63,10 +95,64 @@ jobs:
           make check
     - name: Test Go
       run: go test github.com/snapcore/snapd/...
-  spread:
+  spread-canary:
     needs: unit-tests
-    runs-on: [self-hosted]
+    runs-on: [self-hosted, linux]
+    strategy:
+      fail-fast: false
+      matrix:
+        system:
+        - ubuntu-16.04-64
+        - ubuntu-core-16-64
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Run spread tests
+      run: spread google:${{ matrix.system }}:tests/...
+      # TODO: move this to a reliable post() action.
+    - name: Discard spread workers
+      run: |
+        shopt -s nullglob;
+        for r in .spread-reuse.*.yaml; do
+          spread -discard -reuse-pid="$(echo "$r" | grep -o -E '[0-9]+')";
+        done
+  spread-stable:
+    needs: [unit-tests, spread-canary]
```

Comment:
Doing it this way means we always need to wait for the 40min for spread-canary, right? So each run will take at least time(unittest)+time(canary-spread)+time(spread) ?